File: CS/SpreadsheetControl_WPF_Examples/SpreadsheetActions/CellActions.cs
using System;
using DevExpress.Spreadsheet;
using System.Drawing;
using DevExpress.Xpf.Spreadsheet;
using System.Collections.Generic;

namespace SpreadsheetControl_WPF_Examples
{
    public static class CellActions
    {
        #region Actions
        public static Action<SpreadsheetControl> SelectedCellAction = SelectedCell;
        public static Action<SpreadsheetControl> SetSelectedRangesAction = SetSelectedRanges;
        #endregion

        #region #SelectedCell
        static void SelectedCell(SpreadsheetControl control)
        {
            control.BeginUpdate();

            control.SelectedCell.FillColor = Color.LightGray;
            CellRange c = control.SelectedCell;
            c.FillColor = Color.Blue;

            CellRange currentSelection = control.Selection;
            Formatting rangeFormatting = currentSelection.BeginUpdateFormatting();
            rangeFormatting.Borders.SetOutsideBorders(DevExpress.Utils.DXColor.Green, BorderLineStyle.MediumDashDot);
            currentSelection.EndUpdateFormatting(rangeFormatting);

            control.EndUpdate();
        }
        #endregion #SelectedCell

        #region #SetSelectedRanges
        static void SetSelectedRanges(SpreadsheetControl control)
        {
            control.BeginUpdate();
            Worksheet worksheet = control.ActiveWorksheet;

            CellRange r1 = worksheet.Range["A1:B10"];
            CellRange r2 = worksheet.Range["E12"];
            CellRange r3 = worksheet.Range["D4:E7"];
            List<CellRange> rlist = new List<CellRange>() { r1, r2, r3 };
            control.SetSelectedRanges(rlist);

            control.SelectedCell = worksheet.Cells["E5"];

            control.EndUpdate();
        }
        #endregion #SetSelectedRanges

    }
}

File: VB/SpreadsheetControl_WPF_Examples/SpreadsheetActions/CellActions.vb
Imports System
Imports DevExpress.Spreadsheet
Imports System.Drawing
Imports DevExpress.Xpf.Spreadsheet
Imports System.Collections.Generic

Namespace SpreadsheetControl_WPF_Examples
	Public Module CellActions
		#Region "Actions"
		Public SelectedCellAction As Action(Of SpreadsheetControl) = AddressOf SelectedCell
		Public SetSelectedRangesAction As Action(Of SpreadsheetControl) = AddressOf SetSelectedRanges
		#End Region

		#Region "#SelectedCell"
		Private Sub SelectedCell(ByVal control As SpreadsheetControl)
			control.BeginUpdate()

			control.SelectedCell.FillColor = Color.LightGray
			Dim c As CellRange = control.SelectedCell
			c.FillColor = Color.Blue

			Dim currentSelection As CellRange = control.Selection
			Dim rangeFormatting As Formatting = currentSelection.BeginUpdateFormatting()
			rangeFormatting.Borders.SetOutsideBorders(DevExpress.Utils.DXColor.Green, BorderLineStyle.MediumDashDot)
			currentSelection.EndUpdateFormatting(rangeFormatting)

			control.EndUpdate()
		End Sub
		#End Region ' #SelectedCell

		#Region "#SetSelectedRanges"
		Private Sub SetSelectedRanges(ByVal control As SpreadsheetControl)
			control.BeginUpdate()
			Dim worksheet As Worksheet = control.ActiveWorksheet

			Dim r1 As CellRange = worksheet.Range("A1:B10")
			Dim r2 As CellRange = worksheet.Range("E12")
			Dim r3 As CellRange = worksheet.Range("D4:E7")
			Dim rlist As New List(Of CellRange)() From {r1, r2, r3}
			control.SetSelectedRanges(rlist)

			control.SelectedCell = worksheet.Cells("E5")

			control.EndUpdate()
		End Sub
		#End Region ' #SetSelectedRanges

	End Module
End Namespace
